build.gradle

```diff
@@ -461,46 +455,49 @@ project(':iceberg-aws') {
     api project(':iceberg-api')
     implementation project(':iceberg-common')
     implementation project(':iceberg-core')
-    annotationProcessor "org.immutables:value"
-    compileOnly "org.immutables:value"
-    implementation "com.github.ben-manes.caffeine:caffeine"
+    annotationProcessor libs.immutables.value
+    compileOnly libs.immutables.value
+    implementation libs.caffeine
+    implementation platform(libs.jackson.bom)
     implementation "com.fasterxml.jackson.core:jackson-databind"
     implementation "com.fasterxml.jackson.core:jackson-core"
 
-    compileOnly 'software.amazon.awssdk:url-connection-client'
-    compileOnly 'software.amazon.awssdk:apache-client'
-    compileOnly 'software.amazon.awssdk:auth'
-    compileOnly 'software.amazon.awssdk:s3'
-    compileOnly 'software.amazon.awssdk:kms'
-    compileOnly 'software.amazon.awssdk:glue'
-    compileOnly 'software.amazon.awssdk:sts'
-    compileOnly 'software.amazon.awssdk:dynamodb'
-    compileOnly 'software.amazon.awssdk:lakeformation'
-
-    compileOnly("org.apache.hadoop:hadoop-common") {
+    compileOnly(platform(libs.awssdk.bom))
```

Comment:
I think we should probably do the same here as I mentioned further above for jackson-databind

Reply:
This approach using bom was suggested by @snazy in https://github.com/apache/iceberg/pull/7694#discussion_r1227938340 & https://github.com/apache/iceberg/pull/7694#discussion_r1227943762
which actually has its own advantages